usecases/objects/batch_references_add.go

```diff
@@ -117,6 +127,111 @@ func (b *BatchManager) validateReference(wg *sync.WaitGroup, ref *models.BatchRe
 	}
 }
 
+func validateReferenceMultiTenancy(ctx context.Context,
+	principal *models.Principal, schemaManager schemaManager,
+	repo VectorRepo, source *crossref.RefSource, target *crossref.Ref,
+	tenantKey string,
+) error {
+	if source == nil || target == nil {
+		return fmt.Errorf("can't validate multi-tenancy for nil refs")
+	}
+
+	sourceClass, targetClass, err := getReferenceClasses(
+		ctx, principal, schemaManager, source.Class.String(), target.Class)
+	if err != nil {
+		return err
+	}
+
+	// if both classes have MT enabled but different tenant keys,
+	// no cross-tenant references can be made
+	if sourceClass.MultiTenancyConfig != nil && targetClass.MultiTenancyConfig != nil {
+		if sourceClass.MultiTenancyConfig.TenantKey != targetClass.MultiTenancyConfig.TenantKey {
```

Comment:
Isn't `TenantKey` just a name of property holding actual tenant?
I think validation can not be done on class level, but single objects level.

If `Class1` has `Tenant1` and `Class2` has `Tenant1` and `Tenant2`, then object of `Class1+Tenant1` can be referenced to object of `Class2+Tenant1` but not to `Class2+Tenant2`

Reply:
The validation is still done on object level below this line, but this is a quick short-circuit if these tenant keys do not match.  Then we can exit without parsing objects

Reply:
Agreed, and object validation is included below this line. This is just an early short-circuit that removes the need to parse objects if no reference should be made.

> If Class1 has Tenant1 and Class2 has Tenant1 and Tenant2, then object of Class1+Tenant1 can be referenced to object of Class2+Tenant1 but not to Class2+Tenant2

True, but I believe the multiTenancyConfig.tenantKey should still be matching in this case?

Reply:
does not have to match.
we usually use in our examples or tests `tenantName` as name of tenant property in all classes, that is why they always match.
but that is up to user to name properties holding tenants. could be property named `userName` in `Class1` and `clientName` in `Class2`

Reply:
Ok, this might be worth a discussion, as this could lead to unintended problems for the client application, if this isn't well-documented.

But for now I can remove this check 👍

Reply:
I think, while the tenant keys may not necessarily have the same name, it is essential to ensure that their values match.